<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>681</width>
    <height>410</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Configuration</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_4">
        <item row="2" column="4" colspan="2">
         <widget class="QPushButton" name="chooseCppOutputButton">
          <property name="text">
           <string>Choose Folder...</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Python App Entrypoint:</string>
          </property>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>Open Python Code with:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="4" colspan="2">
         <widget class="QPushButton" name="chooseAppEntryButton">
          <property name="text">
           <string>Choose File...</string>
          </property>
         </widget>
        </item>
        <item row="4" column="3">
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>Open C++ Code with:</string>
          </property>
         </widget>
        </item>
        <item row="4" column="4" colspan="2">
         <widget class="QComboBox" name="openCppCodeWithCombo">
          <item>
           <property name="text">
            <string>Visual Studio</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>C++ Code Ouptut Folder:</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1" colspan="3">
         <widget class="QLineEdit" name="cppCodeOutputFolderLine"/>
        </item>
        <item row="1" column="1" colspan="3">
         <widget class="QLineEdit" name="runtimeTypingFolderLine"/>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Runtime Typing Folder:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1" colspan="3">
         <widget class="QLineEdit" name="pythonAppEntrypointLine"/>
        </item>
        <item row="1" column="4" colspan="2">
         <widget class="QPushButton" name="chooseRuntimeTypingButton">
          <property name="text">
           <string>Choose Folder...</string>
          </property>
         </widget>
        </item>
        <item row="8" column="0">
         <widget class="QGroupBox" name="cppStandardLibraryGroup">
          <property name="title">
           <string>C++ Standard Library</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_3">
           <item row="0" column="0">
            <widget class="QRadioButton" name="qtStandardLibRadio">
             <property name="text">
              <string>Qt</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QRadioButton" name="stlStandardLibRadio">
             <property name="text">
              <string>STL</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="7" column="0">
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Orientation::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="4" column="1" colspan="2">
         <widget class="QComboBox" name="openPythonCodeWithCombo">
          <item>
           <property name="text">
            <string>Wing</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>Runtime Type Check</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="0" colspan="3">
         <widget class="QTableWidget" name="runtimeTypeCheckTable">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="sortingEnabled">
           <bool>true</bool>
          </property>
          <attribute name="horizontalHeaderStretchLastSection">
           <bool>true</bool>
          </attribute>
          <attribute name="verticalHeaderStretchLastSection">
           <bool>false</bool>
          </attribute>
          <column>
           <property name="text">
            <string>Local ID</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Scope ID</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Editor Link</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Assigned Types</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>C++ Type</string>
           </property>
          </column>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QPushButton" name="pushButton_2">
          <property name="text">
           <string>Start Type Check</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QPushButton" name="pushButton_3">
          <property name="text">
           <string>Reset Type Check</string>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Orientation::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_3">
       <attribute name="title">
        <string>C++ Code Generation</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_5">
        <item row="2" column="0">
         <widget class="QPushButton" name="pushButton">
          <property name="text">
           <string>Generate C++ Code</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Orientation::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="1" column="0" colspan="2">
         <widget class="QTableWidget" name="tableWidget">
          <property name="sortingEnabled">
           <bool>true</bool>
          </property>
          <attribute name="horizontalHeaderStretchLastSection">
           <bool>true</bool>
          </attribute>
          <column>
           <property name="text">
            <string>Python Side</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>C++ Side</string>
           </property>
          </column>
          <column>
           <property name="text">
            <string>Conversion Result</string>
           </property>
          </column>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>681</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuProject">
    <property name="title">
     <string>Project</string>
    </property>
    <addaction name="actionNewProject"/>
    <addaction name="actionSaveProject"/>
    <addaction name="actionExitProject"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
    <addaction name="actionGitHub_Issues_Forum"/>
    <addaction name="actionLicense"/>
    <addaction name="actionCheck_for_Updates"/>
   </widget>
   <addaction name="menuProject"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionNewProject">
   <property name="text">
    <string>New</string>
   </property>
  </action>
  <action name="actionSaveProject">
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="actionExitProject">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionGitHub_Issues_Forum">
   <property name="text">
    <string>GitHub Issues Forum</string>
   </property>
  </action>
  <action name="actionLicense">
   <property name="text">
    <string>License</string>
   </property>
  </action>
  <action name="actionCheck_for_Updates">
   <property name="text">
    <string>Check for Updates</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
